src/PooledArrays.jl

```diff
@@ -139,6 +140,8 @@ _widen(::Type{Int32}) = Int64
 Freshly allocate `PooledArray` using the given array as a source where each
 element will be referenced as an integer of the given type.
 
+`PooledArray` constructor is not type stable.
```

Comment:
How about explaining why that's the case, making the link with the next paragraph? And isn't the constructor type stable when `reftype` is specified?

BTW, maybe we should mark the constructor as `@inline` to ensure that when the default value of keyword arguments is used, the return type is inferred?

Reply:
I have added inline. I have rewritten the docstring. Indeed when `reftype` is passed the constructor is type stable.